### DISCUSSION TITLE
Program ATTiny828

### DISCUSSION THREAD

**Discussion by vBakaMO** [SpenceKonde/ATTinyCore#548] [General]
Hey; so I am starting out with this Library, and have one question- how do I program an ATTiny828 connected to a custom PCB? Am I able to connect it to an Arduino as a programmer and do it that way, and if so, how?

**Comment by grandaspanna** [comment]
This is the wrong core for the ATTiny828 - it's close, but you need this one: https://github.com/SpenceKonde/ATTinyCore

You can use an Arduino as a programmer, but best ask over in that repository :-)

**Comment by SpenceKonde** [comment]
Transferred to the ATTinyCore repo. 

The 828 can be programmed like any other classic AVR - ISP programmer connected to the SPI pins. A well designed custom board would include a 6-pin ISP header. 

That said, as pointed out, the 828 is not one of the "modern AVR" tinyAVR devices (that is, it is a tinyAVR that is not a 0-series, 1-series, or 2-series). Worse still, as far as I can tell,, there was some horrible debacle during the development process that left it with it's crown-jewel largely defanged - the register layout clearly implies that it was meant to have the snazzy ADC that the 841 got - only with tons of channels. Instead, it wound up with a "basic" ADC - 1.1v reference voltage, single-ended only - but 28 input channels. It's also got that wonky pin that's pulled down if WDT isn't enabled due to silicon bug. The datasheet looks to have been hastily (read: poorly) scrubbed of references to the differential ADC that it either was released without, or which didn't work and was removed from the docs (and a half-working differential ADC is actually in the released silicon). 

So if the custom board does not yet exist, I would council against designing it with the 828. Did you mistake it for one of the new tinyAVR 2-series parts?
